packages/file-search/src/browser/quick-file-open.ts

```diff
@@ -126,79 +130,76 @@ export class QuickFileOpenService implements QuickOpenModel {
         this.cancelIndicator.cancel();
         this.cancelIndicator = new CancellationTokenSource();
         const token = this.cancelIndicator.token;
-        const proposed = new Set<string>();
+        
         const rootUri = workspaceFolder.uri;
-        const handler = async (result: string[]) => {
-            if (!token.isCancellationRequested) {
-                const root = new URI(rootUri);
-                result.forEach(p => {
-                    const uri = root.withPath(root.path.join(p)).toString();
-                    proposed.add(uri);
-                });
-                const itemPromises = Array.from(proposed).map(uri => this.toItem(uri));
-                acceptor(await Promise.all(itemPromises));
+        const root = new URI(rootUri);
+        const alreadyCollected = new Set<string>();
+        
+        const recentlyUsedItems: QuickOpenItem[] = [];
+        const locations = this.navigationLocationservice.locations()
+        if (locations.length > 0) {
+            for (const location of locations) {
+                const uriString = location.uri.toString();
+                if (!alreadyCollected.has(uriString) && fuzzy.test(lookFor, uriString)) {
+                    recentlyUsedItems.push(await this.toItem(location.uri, recentlyUsedItems.length === 0 ? 'recently used' : undefined));
```

Comment:
I am not sure if this (`recentlyUsedItems.length === 0`) is correct. At least I can see a glitch. Sometimes the `recently used` is missing.
![recently used is missing](https://user-images.githubusercontent.com/1405703/43950133-cafccd68-9c8f-11e8-8f21-df0b14b0c0ef.gif)

Can we have `recently opened` instead of `recently used`? Then the behavior would be in sync with VS Code.

Reply:
This is a bit tricky, as we want to rely on the fuzzy match functionality of widget. It might filter out the element, that contains the group label. I improved it by using 'fuzzy' and filter the elements based on that, too. But it seems like the algorithm is slightly different, so that sometimes an element doesn't get filtered by fuzzy but by the fuzzyMatch functionality of the quickopen widget. 
If that element is holding the group label it will be filtered out.